src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -2432,7 +2440,11 @@ ConcreteClusterPath path(static_cast<chip::EndpointId>([endpointID unsignedShort
 
 - (BOOL)isEqualToClusterPath:(MTRClusterPath *)clusterPath
 {
-    return [_endpoint isEqualToNumber:clusterPath.endpoint] && [_cluster isEqualToNumber:clusterPath.cluster];
+    if (!clusterPath)
+        return NO;
+
+    return (clusterPath.endpoint && [_endpoint isEqualToNumber:clusterPath.endpoint])
+        && (clusterPath.cluster && [_cluster isEqualToNumber:clusterPath.cluster]);
```

Comment:
Here, the API claims none of these things are nullable.  How would they be nil?  And any use of these things if they are nil would blow up in Swift, I expect.

If we want to protect against this, we should be failing init with nil to start with.  But again, those things are not supposed to be nullable even there!